android/src/main/java/co/uk/hive/reactnativegeolocation/RNGeolocationModule.java

```diff
@@ -87,29 +116,82 @@ public void isLocationEnabled(Promise promise) {
     private <T> Function<T, Object> convertCallback(Callback callback) {
         return t -> {
             if (t instanceof ApiException) {
-                final ApiException apiException = (ApiException) t;
-                final WritableMap errorMap = Arguments.createMap();
-
-                errorMap.putString("exceptionType", t.getClass().toString());
-                errorMap.putString("message", apiException.getMessage());
-                errorMap.putString("code", String.valueOf(apiException.getStatusCode()));
-                errorMap.putString("JIRA", "MA1-1200");
-
-                callback.invoke(errorMap);
+                handleApiException((ApiException) t, callback);
             } else if (t instanceof Exception) {
                 final Exception exception = (Exception) t;
                 final WritableMap errorMap = Arguments.createMap();
-
                 errorMap.putString("exceptionType", t.getClass().toString());
                 errorMap.putString("message", exception.getMessage());
                 errorMap.putString("JIRA", "MA1-1200");
-
                 callback.invoke(errorMap);
             } else {
                 callback.invoke(t);
             }
-
             return null;
         };
     }
+
+    private void handleApiException(ApiException apiException, Callback failureCallback) {
+        switch (apiException.getStatusCode()) {
+            case LocationSettingsStatusCodes.RESOLUTION_REQUIRED: {
+                // Location settings are not satisfied. But could be fixed by showing the user a dialog.
+                try {
+                    // Cast to a resolvable exception.
+                    final ResolvableApiException resolvable = (ResolvableApiException) apiException;
```

Comment:
For Android 10 and above, if the settings check fails, this checks to see if the the issue is resolvable and presents a dialog allowing the user to enable high accuracy directly..